### ISSUE TITLE
Decouple serialization for SCM and Inventory updates

### ISSUE THREAD

**Issue by radioactive73** [ansible/awx#6956]
ISSUE TYPE
 - Feature Idea

SUMMARY
Decouple serialization for SCM and Inventory updates - I have so many jobs running from AWX now and some are long running (up to 2 hours), that an SCM or Inventory update can cause significant bottlenecks.

The SCM / Inventory update itself doesn't take long but if it get's kicked off while a long running job is active then it get's queued, and because it get's queued, every other job that's ran after it get's queued too when normally they would run in parallel (concurrently) with the long running job.

My inventory get's automatically updated when kickstarting a new server, as it needs to in order for the provision playbook to find the server. And my SCM get's updated automatically when I push a change to my GIT repo. Technically "Update on Launch" is enabled but only as a fall-back, the cache is set for 24hours and it needs to update on the two events described (GIT push and Kickstart). Since those two events occur regularly this bottleneck is being experienced fairly regularly and I only see it getting worse as AWX handles more jobs. The server resource wise is fine so it can handle more jobs, and it can always scale up more if needed.

Perhaps something like a shadow (temporary copy or cached) inventory / SCM can be used on launch so the original is decoupled and can be updated without affecting the running job? If the repo is too big maybe just the dependencies for the job? Anyways, I won't pretend to know how difficult this would be to implement so I'll leave that up to you guys. Even just having the inventory decoupled would be a win but an option to do this or built in by default would help out a lot! I'm sure other people and organizations would benefit from the gained scalability too.

**Comment by wenottingham** [comment]
What version of AWX are you on? Both of these have been decoupled in more recent versions.

**Comment by radioactive73** [comment]
I missed that in the release notes, I'll test the latest version (next week hopefully) and see if it's still an issue but based on the descriptions it looks like both issues were resolved. Actually, I only found the inventory sync in the release notes but found an issue that was merged for SCM.

**Comment by wenottingham** [comment]
OK, closing. Reopen/file a new issue if you hit something.
